### ISSUE TITLE
Errors are never displayed (#381)

### ISSUE THREAD

**Issue by kenygamer** [discord-php/DiscordPHP#428]
Reopening because the errors are now displaying in events and timers but not in `ExtendedPromiseInterface->done()`, etc.

**Comment by kenygamer** [comment]
To note the importance of this, issues like #425 are harder to fix without any error logs.

**Comment by davidcole1340** [comment]
Can you show an example? Works for me and no one else has had a problem yet

**Comment by kenygamer** [comment]
I haven't had it however when I find it again I'll reopen this issue.

**Comment by sudofox** [comment]
I'm also having an issue with this. What would normally show an error (such as an undefined function, a database query not running correctly, and so on) doesn't show at all in the CLI output. I've tried setting error reporting to E_ALL and so on. Most of what would trigger an error happens inside the `$discord->on('message', ...)` event handler. I'm glad it doesn't crash on error, but I'm often left dropping in `error_log`s and moving them line by line until I find the line that doesn't finish executing (and thus doesn't trigger our error_log).

**Comment by lessevv** [comment]
Did you check your server settings? You can set the display_errors option within your php.ini to show the errors. They all show for me in the CLI. The fact they are being outputted in the error_log file, also indicates this is something you need to fix on server level.

If you can't do this, try to set `ini_set('display_errors', 'On')` in your file.

**Comment by sudofox** [comment]
I just figured out how to handle it.

First, after digging even further through the various issues, I found that our version of discordphp was just slightly out of date and exception handling was changed - now errors etc will crash the bot.

I added a `try { ... } catch { ... }` around the entire message handler function and it now logs the errors in the CLI output. (Would've been nice to get it so that I can still `$message->reply` with "an error occured" from within the catch block but that does not seem to work or show any success/failure despite $message being instantiated still. Ah well)

```php
function handleMessage($message) {

    try {
    -snip-
    } catch (Error $e) {
        $loggedError = "Error caught: " . $e->getMessage() . PHP_EOL . $e->getTraceAsString();
        error_log($loggedError);
        $message->reply($loggedError); // no worky, but meh
    }
}
```
